src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -252,26 +259,56 @@ export class DataSourceSavedObjectsClientWrapper {
 
   private async validateAttributes<T = unknown>(attributes: T) {
     const { title, endpoint, auth } = attributes;
-    if (!title?.trim?.().length) {
+
+    this.validateTitle(title);
+    await this.validateEndpoint(endpoint, attributes as DataSourceAttributes);
+    await this.validateAuth(auth);
+  }
+
+  private validateTitle(title: string) {
+    if (!title.trim().length) {
       throw SavedObjectsErrorHelpers.createBadRequestError(
         '"title" attribute must be a non-empty string'
       );
     }
 
+    if (title.length > DATA_SOURCE_TITLE_LENGTH_LIMIT) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        `"title" attribute is limited to ${DATA_SOURCE_TITLE_LENGTH_LIMIT} characters`
+      );
+    }
+  }
+
+  private async validateEndpoint(endpoint: string, attributes: DataSourceAttributes) {
     if (!isValidURL(endpoint, this.endpointBlockedIps)) {
       throw SavedObjectsErrorHelpers.createBadRequestError(
         '"endpoint" attribute is not valid or allowed'
       );
     }
+    try {
+      const dataSourceClient: OpenSearchClient = await this.dataSourcesService.getDataSourceClient({
+        savedObjects: {} as any,
+        cryptography: this.cryptography,
+        testClientDataSourceAttr: attributes as DataSourceAttributes,
+        authRegistry: await this.authRegistryPromise,
+        customApiSchemaRegistryPromise: this.customApiSchemaRegistryPromise,
+      });
```

Comment:
Thanks for this change.

Are there any chances we can reuse this function: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source_management/public/components/utils.ts#L206?

Since datasource and datasource manager plugin will both exist in the same time, so the validation path in datasource manager plugin should always been routed. 

Then we depend on sending request to `/internal/data-source-management/validate` to test connection instead of creating a new datasource client.

What do you think?

Reply:
Do you mean when user click create data source button, we first call `/internal/data-source-management/validate` to validate and then call create data source API? @xinruiba

Reply:
Yes

Pros:

- We rely on the URL for data source validation.

Cons:

- If customers directly create a data source by calling the URL, the data source validation will be bypassed.

I'll leave the decision to you~

Reply:
Sounds good to me. Approving~

Reply:
Non blocking comment.

Maybe a question to designer. Since we already do validation while datasource creation. Do we sill need a separate test connection?

Reply:
good point, @kgcreative @xeniatup, I wonder what's your thought on this

Reply:
Given the updated validation, i don't think we need it in the create page.